src/Driver/CompilerCache.php

```diff
@@ -89,7 +89,12 @@ public function compile(QueryParameters $params, string $prefix, FragmentInterfa
      */
     protected function hashInsertQuery(QueryParameters $params, array $tokens): string
     {
-        $hash = 'i_' . $tokens['table'] . implode('_', $tokens['columns']) . '_r' . ($tokens['return'] ?? '');
+        $hash = \sprintf(
+            'i_%s%s_r%s',
+            $tokens['table'],
+            \implode('_', $tokens['columns']),
+            \implode('_', (array)($tokens['return'] ?? []))
```

Comment:
Is it tested with fragments?

Reply:
Tested, but I added a standalone unit test